man/systemd.resource-control.xml

```diff
@@ -686,6 +686,18 @@
           TTYs and all ALSA sound devices,
           respectively. <literal>char-cpu/*</literal> is a specifier
           matching all CPU related device groups.</para>
+
+          <para>Note that whitelists defined this way should only reference device groups which are
+          resolvable at the time the unit is started. Any device groups not resolvable then are not added to
+          the device whitelist. In order to work around this limitation, consider extending service units
+          with an <command>ExecStartPre=/sbin/modprobe…</command> line that loads the necessary
+          kernel module implementing the device group if missing. Example: <programlisting>…
+[Service]
+ExecStartPre=-/sbin/modprobe -abq loop
+DeviceAllow=block-loop
+DeviceAllow=/dev/loop-control
```

Comment:
Shouldn't we instead implicitly modprobe `devname:loop-control` and other `devname:...` as appropriate instead?

I also find this description missing crucial details. What does "all ALSA sound devices" mean? Is `/dev/snd/seq` included or not?

Reply:
see my comments on the original bug about this. The problem is not about the /dev/ device nodes (which are created by tmpfiles.d/ from the kmod data early on, and hence stat()able, and hence resolvable just fine for the purpose of the DeviceAllow= stuff), the problem is about expressions like `block-loop` which are resolved via /proc/devices, and that file only lists registered subsystems, i.e. requires loop.ko to be loaded for block-loop to be resolvable...

And yeah, /dev/snd/seq is included, since it shows up in /proc/devices with its major (i.e. all of alsa has a single major after all)

Reply:
OK. After rereading the existing docs and your patch, I think it *does* say all the right things. Maybe it could be improved to be made a little clearer, e.g. by ordering the /dev/* stuff along with examples first, and then block-* and char-* along with examples second.

Should we ask the kernel to provide modaliases for those names (char:alsa), (char:loop)?